test_util/test_upgrade_vpc.py

```diff
@@ -173,9 +172,19 @@ def create_marathon_dns_app(app_id: str, healthcheck_app_id: str) -> dict:
     # DNS resolution app to make sure DNS is available during the upgrade.
     # Periodically resolves the healthcheck app's domain name and logs whether
     # it succeeded to a file in the Mesos sandbox.
+
+    dns_parsed_app_name = marathon_app_id_to_mesos_dns_subdomain(healthcheck_app_id)
+    resolve_name = '{}.marathon.agentip.dcos.thisdcos.directory'.format(dns_parsed_app_name)
+
     return {
         "id": '/' + app_id,
         "cmd": """
+until host -W $TIMEOUT_SECONDS $RESOLVE_NAME
+do
+    echo "Waiting to resolve $RESOLVE_NAME for the first time..."
+    sleep 1
+done
```

Comment:
Hmm another option is to keep the mesos dns http api check, but just check a different dns address than the one that we're resolving (both dns addresses point to the same app).

The reason I originally removed the check was because i didn't want to create 2 separate dns addresses, one that is in mesos-dns and another that we're actually testing. But it seems that that way is actually preferable to this.